analyzer/codechecker_analyzer/cmd/parse.py

```diff
@@ -825,7 +849,7 @@ def skip_html_report_data_handler(report_hash, source_file, report_line,
                               context.path_plist_to_html_dist,
                               skip_html_report_data_handler,
                               html_builder,
-                              trim_path_prefixes_handler)
+                              TrimPathPrefixHandler(trim_path_prefixes))
```

Comment:
Why is it better to use this new class here than the original solution (using a simple function)?

Reply:
When I tried to understand what the main function does, I recognized that this function is too big. To reduce its complexity I moved the nested trim_path_prefixes_handler() function outside of the main(). But outside of this function the trim_path_prefixes variable does not in the scope, so I should change type of the callback from a function to a callable object.
This modification is a refactor to make easier understand of this code.